### PR TITLE
PCM_API

### PR DESCRIPTION
**Pull request by FloH11110**
open points:
- used SPI implementation not finished
- address for CSEL is dummy
- init setup with start values

### CODE REVIEW

File: source/PCM_API.c [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+void pcm_init(void){
+    PCM_LOG("PCM init\n");
+    
+    // todo !!!!!!!!!!
+}
+
+uint8_t send_receive(uint8_t address, uint8_t data){
+    uint8_t data_send[2] = {address, data};
+    uint8_t data_receive[2] = {0, 0};
+
```
Comment (SebastianWolfarth) [inline source/PCM_API.c:32]: Tip:
Format Adress and Data as hex (0x%x)
I think its easier to interpret the log.
But its up to you.

File: source/PCM_API.c [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+    
+    send_receive(PCM_register_18_w, data);
+}
+
+audio_data_format_t get_audio_data_format(void){
+    uint8_t data = send_receive(PCM_register_18_r, 0x00);
+    uint8_t bit_4 = (data >> 4) & 1;
+    uint8_t bit_5 = (data >> 5) & 1;
+    uint8_t bit_6 = (data >> 6) & 1;
+    return (4 * bit_6 + 2 * bit_5 + bit_4);
```
Comment (SebastianWolfarth) [inline source/PCM_API.c]: little bit complicated?
return (data>>4) & 0x7;
would do the same??

File: source/PCM_API.c [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+    }
+    
+    send_receive(PCM_register_20_w, data);
+}
+
+oversampling_rate_t get_oversampling_rate(void){
+    uint8_t data = send_receive(PCM_register_20_r, 0x00);
+    uint8_t bit_0 = (data >> 0) & 1;
+    uint8_t bit_1 = (data >> 1) & 1;
+    return (2 * bit_1 + bit_0);
```
Comment (SebastianWolfarth) [inline source/PCM_API.c]: same here

File: source/PCM_API.c (hunk 2 above)
Comment (FloH11110) [inline source/PCM_API.c]: done

File: source/PCM_API.c (hunk 3 above)
Comment (FloH11110) [inline source/PCM_API.c]: done

File: source/PCM_API.c (hunk 1 above)
Comment (FloH11110) [inline source/PCM_API.c:32]: done

**Comment by FloH11110** [comment]
have done a few updates like commented before

**Comment by FloH11110** [comment]
have done a few updates as commented before

**Pull request merged**